### PR TITLE
Add ability to reconnect to games

### PR DESCRIPTION
**Pull request by czinn**
Currently works if app crashes and then player restarts app

Should probably work if server crashes and restarts (or if network connection goes away and comes back) but right now the app crashes if the app loses connection so I haven't tested this yet

### CODE REVIEW

**Comment by czinn** [comment]
@neynt Please review

File: server/adrestia_database.cc [hunk 1]
```diff
@@ -815,8 +755,8 @@ json adrestia_database::register_new_account_in_database(
     pqxx::work work(psql_connection);
     try {
       run_query(logger, work, R"sql(
-        INSERT INTO adrestia_accounts (uuid, user_name, tag, hash_of_salt_and_password, salt)
-        VALUES (%s, %s, %s, %s, %s)
+        INSERT INTO adrestia_accounts (uuid, user_name, tag, hash_of_salt_and_password, salt, last_login, last_message)
```
Comment (Takareer) [inline server/adrestia_database.cc]: Last login/last message could be a lot of writes. We could have a separate table storing this to keep adrestia_accounts relatively "clean", but I'm not sure if it is worthwhile...

File: server/pushers/push_active_games.cc [hunk 2]
```diff
@@ -42,27 +42,53 @@ std::vector<json> adrestia_networking::PushActiveGames::push(const Logger &logge
...
+   *                        rules that aren't the most recent rules, the game's
+   *                        rules will be included in the first update.
    *           Note that it follows that reported game_views are waiting for
    *           the player to make a move.
    */
 
   // Check for games
-  pqxx::connection psql_connection = adrestia_database::establish_connection();
-  json active_games = adrestia_database::check_for_active_games_in_database(logger, psql_connection, uuid);
+  adrestia_database::Db db(logger);
```
Comment (Takareer) [inline server/pushers/push_active_games.cc:79]: My beautiful monolith... torn apart... (jk this is a good idea)

File: server/pushers/push_active_games.cc [hunk 3]
```diff
@@ -42,27 +42,53 @@ std::vector<json> adrestia_networking::PushActiveGames::push(const Logger &logge
    *             MESSAGE_KEY: "You have [new|changed|new and changed|] games!" (depending on situation)
    *             "updates": A list of updated game views or states, of the form
```
Comment (Takareer) [inline server/pushers/push_active_games.cc:65]: You have added keys game_rules and player_move, so you should put them here

File: server/schema.sql [hunk 4]
```diff
@@ -5,6 +5,8 @@ CREATE TABLE IF NOT EXISTS adrestia_accounts (
 	tag VARCHAR NOT NULL,
 	hash_of_salt_and_password BYTEA NOT NULL,
 	salt VARCHAR NOT NULL,
+	last_login TIMESTAMP,
```
Comment (Takareer) [inline server/schema.sql]: Use TIMESTAMPTZ, not TIMESTAMP. Non-zoned timestamps are fabrications of Nurgle, the Chaos God of decay and disease.

File: server/adrestia_database.cc [hunk 5]
```diff
@@ -304,7 +304,7 @@ json adrestia_database::retrieve_player_info_from_database (
    * @returns json containing the following keys:
    *          "player_id": The player id in the game, as integer
    *          "player_state": The player state in the game, as integer
-   *          "player_move": The player move, if any, as a string. Empty if null.
+   *          "player_move": The player move, if any, as a string. Null if null.
```
Comment (Takareer) [inline server/adrestia_database.cc]: player_move is now JSON, not a string

File: server/adrestia_database.cc (hunk 1 above)
Comment (czinn) [reply server/adrestia_database.cc]: It won't be less writes in another table, and other than this, there should be relatively few writes to the adrestia_accounts table, so I don't think we'd get any better results by splitting up the data.

**Comment by neynt** [comment]
LGTM

**Pull request merged**